xwiki-platform-core/xwiki-platform-store/xwiki-platform-store-filesystem-oldcore/src/main/java/org/xwiki/store/legacy/store/internal/FilesystemAttachmentVersioningStore.java

```diff
@@ -134,7 +134,7 @@ XWikiAttachmentArchive loadArchive(final XWikiAttachment attachment, final Attac
             attach.setAttachment_content(new FilesystemAttachmentContent(contentFile, attach));
             attach.setContentStore(FileSystemStoreUtils.HINT);
             // Pass the document since it will be lost in the serialize/deserialize.
-            attach.setDoc(attachment.getDoc());
+            attach.setDoc(attachment.getDoc(), false);
```

Comment:
After reading the chat discussion, the change looks ok.
Though, you probably planned that, but I suggest at least adding a test that would fail without this fix.

Reply:
So this is the case with the new integration test I provided.